src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/WSSecurityXXX2005.cs

```diff
@@ -0,0 +1,244 @@
+                        if (tokenTypeUri != null)
+                        {
+                            writer.WriteAttributeString(CoreWCF.XD.SecurityXXX2005Dictionary.Prefix.Value, CoreWCF.XD.SecurityXXX2005Dictionary.TokenTypeAttribute, CoreWCF.XD.SecurityXXX2005Dictionary.Namespace, tokenTypeUri);
+                        }
+
+                        this.StrEntries[i].WriteContent(writer, keyIdentifierClause);
+                        writer.WriteEndElement();
+                        return;
+                    }
+                }
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.StandardsManagerCannotWriteObject, keyIdentifierClause.GetType())));
+            }
+
+            private string GetTokenTypeUri(StrEntry str, SecurityKeyIdentifierClause keyIdentifierClause)
+            {
+                bool emitTokenType = this.EmitTokenType(str);
+                if (emitTokenType)
+                {
+                    string tokenTypeUri;
+                    if (str is LocalReferenceStrEntry)
+                    {
+                        tokenTypeUri = (str as LocalReferenceStrEntry).GetLocalTokenTypeUri(keyIdentifierClause);
+                        // only emit token type for SAML,Kerberos and Encrypted References
+                        switch (tokenTypeUri)
+                        {
+                            case SecurityXXX2005Strings.Saml20TokenType:
+                            case SecurityXXX2005Strings.SamlTokenType:
+                            case SecurityXXX2005Strings.EncryptedKeyTokenType:
+                            case SecurityJan2004Strings.KerberosTokenTypeGSS: break;
+
+                            default:
+                                tokenTypeUri = null;
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        tokenTypeUri = str.GetTokenTypeUri();
+                    }
+
+                    return tokenTypeUri;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+
+            private bool EmitTokenType(StrEntry str)
+            {
+                bool emitTokenType = false;
+                // we emit tokentype always for SAML and Encrypted Key Tokens 
+                if (
+                    //(str is SamlJan2004KeyIdentifierStrEntry)
+                        //||
+                        (str is EncryptedKeyHashStrEntry)
+                       // || (str is SamlDirectStrEntry)
+                       )
+                {
+                    emitTokenType = true;
+                }
+                else if (this.EmitBspRequiredAttributes)
+                {
+                    if (
+                        //(str is KerberosHashStrEntry)
```

Comment:
You probably want to uncomment this to enable the Windows auth scenario

Reply:
will take care in Window Auth